### PR TITLE
Removed how to use thread pool topics

### PR DESCRIPTION
**Pull request by pkulikov**
Per [discussion](https://github.com/dotnet/docs/issues/6180#issuecomment-407232277) in #6180 and since PRs dotnet/samples#206 and dotnet/dotnet-api-docs#389 are merged, this PR removes "How to use a thread pool" topics from the conceptual docs and redirects to the [ThreadPool.QueueUserWorkItem](https://docs.microsoft.com/en-us/dotnet/api/system.threading.threadpool.queueuserworkitem?view=netframework-4.7.2) API page.

Contributes to #6180

### CODE REVIEW

**Review by jzeferino — APPROVED** [review]
LGTM, the link is working.

File: docs/standard/threading/the-managed-thread-pool.md [hunk 1]
```diff
@@ -80,7 +80,11 @@ The <xref:System.Threading.ThreadPool> class provides your application with a po
...
+ Beginning with the .NET Framework 4, the easiest way to use the thread pool is to use the [Task Parallel Library (TPL)](../parallel-programming/task-parallel-library-tpl.md). By default, parallel library types like <xref:System.Threading.Tasks.Task> and <xref:System.Threading.Tasks.Task%601> use thread pool threads to run tasks.
```
Comment (rpetrusha) [inline docs/standard/threading/the-managed-thread-pool.md]: nit unrelated to your PR: I wouldn't describe the TPL as "parallel library types"; I think I would change this to "TPL types like..."

File: docs/standard/threading/the-managed-thread-pool.md [hunk 2]
```diff
@@ -80,7 +80,11 @@ The <xref:System.Threading.ThreadPool> class provides your application with a po
...
+
+You can also use the thread pool by calling <xref:System.Threading.ThreadPool.QueueUserWorkItem%2A?displayProperty=nameWithType> from managed code (or `CorQueueUserWorkItem` from unmanaged code) and passing a <xref:System.Threading.WaitCallback> delegate representing the method that performs the task.
```
Comment (rpetrusha) [inline docs/standard/threading/the-managed-thread-pool.md]: again unrelated to your PR: CoreQueueUserWorkItem should be described as ICorThreadpool::CorQueueUserWorkItem and can be a link (../../framework/unmanaged-api/hosting/icorthreadpool-corqueueuserworkitem-method.md)

**Review by rpetrusha — CHANGES_REQUESTED** [review]
This looks really good, @pkulikov. Thanks for doing this work. I've left two comments and suggested changes that are unrelated to your PR.

**Comment by pkulikov** [comment]
@rpetrusha thank you for review; I've addressed both of your comments.

**Review by rpetrusha — APPROVED** [review]
Thanks, @pkulikov, for all the work you've put into streamlining the docs, and also for making the additional set of changes. I'll merge your PR now.

**Pull request merged**